File: selfstudy/python/transactions_reverse.py
'''
task 138:
Transactions - Reverse
You need to reverse the order of transaction amounts in dollars
'''

def reverse_order_transactions(amounts):
    return amounts[::-1]

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], reverse_order_transactions([10, 20, 30])),
        ([50, 60, 70], reverse_order_transactions([50, 60, 70])),
        ([1, 2, 3], reverse_order_transactions([1, 2, 3])),
        ([100, 200], reverse_order_transactions([100, 200])),
        ([], reverse_order_transactions([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = reverse_order_transactions(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# works i think

File: selfstudy/js/transactions_reverse.js
/*
task 38:
Transactions - Reverse
You need to reverse the order of transaction amounts in dollars
*/

function reverse_order_transactions(amounts) {
  return [...amounts].reverse();
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: reverse_order_transactions([10, 20, 30]) },
  { input: [50, 60, 70], expected: reverse_order_transactions([50, 60, 70]) },
  { input: [1, 2, 3], expected: reverse_order_transactions([1, 2, 3]) },
  { input: [100, 200], expected: reverse_order_transactions([100, 200]) },
  { input: [], expected: reverse_order_transactions([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('reverse_order_transactions', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(reverse_order_transactions(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, reverse_order_transactions(t.input) === t.expected || JSON.stringify(reverse_order_transactions(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done
